### PR TITLE
Added setting for overriding SESSION_ENGINE

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?
Related to #1008 

#### What's this PR do?
Changes the `SESSION_ENGINE` setting to be configurable via env var

### CODE REVIEW

File: main/settings.py [hunk 1]
```diff
@@ -185,7 +185,15 @@
...
 
-SESSION_ENGINE = "django.contrib.sessions.backends.signed_cookies"
+SESSION_ENGINE = get_string(
+    "SESSION_ENGINE",
+    "django.contrib.sessions.backends.signed_cookies",
+    description=(
+        "The backend that supports user sessions "
+        "(https://docs.djangoproject.com/en/3.1/topics/http/sessions/#configuring-the-session-engine)"
+    ),
+    required=True,
```
Comment (rhysyngsun) [inline main/settings.py:195]: Can you make this `False` instead? It'll cause the app to fail to boot otherwise and there's a reasonable default.

Comment (gsidebo) [reply main/settings.py]: Misunderstanding on my end. Fixed

**Comment by blarghmatey** [comment]
Should we make this a dictionary mapping? So, rather than the full module path we populate the env var with values such as `cookie`, `cache`, `db`?

**Comment by gsidebo** [comment]
@blarghmatey I don't feel too strongly either way. I think we're only somewhat more likely to have a typo or something in the full module path than just the single module name

**Comment by blarghmatey** [comment]
I don't have a strong conviction on that, just a suggestion to improve ergonomics to prevent having to look up the possible values. Might be worth at least adding a comment with some of the options, or a link to relevant docs?

**Comment by gsidebo** [comment]
@blarghmatey Just updated with that suggestion and possible values

**Review by rhysyngsun — APPROVED** [review]

**Comment by blarghmatey** [comment]
Looks good to me

**Pull request merged**